File: project/src/main/java/com/isa/project/service/FishingLessonService.java
package com.isa.project.service;

import java.util.List;

import com.isa.project.dto.DateSearchDTO;
import com.isa.project.dto.FishingLessonDTO;
import com.isa.project.dto.ReservationSearchDTO;
import com.isa.project.dto.SearchParamsDTO;
import com.isa.project.dto.SortDTO;
import com.isa.project.model.FishingLesson;

public interface FishingLessonService {

	public FishingLesson createFishingLesson(FishingLessonDTO dto);
	public List<FishingLesson> getByInstructorId(Long instructorId);
	public FishingLesson getById(Long id);
	public FishingLesson getByIdForInstructor(Long id);
	public FishingLesson editFishingLesson(FishingLessonDTO dto);
	public List<FishingLesson> getAll();
	public List<FishingLesson> sort(SortDTO dto);
	public List<FishingLesson> search(String searchTerm);
	public List<FishingLesson> getAvailableLessons(ReservationSearchDTO dto);
	public List<FishingLesson> lessonsAvailableForCertainDate(DateSearchDTO dto);
	public List<FishingLesson> searchForInstructor(String searchTerm, Long id);
	public List<FishingLesson> getLessonsByClientSubscription(Long clientId);
	public List<FishingLesson> sortAvailableLessons(ReservationSearchDTO dto);
	public List<FishingLesson> searchByMoreParams(SearchParamsDTO dto);
	public List<FishingLesson> getAvailableLessonsForInstructor(ReservationSearchDTO dto);
}

File: front/isa-front/src/app/services/fishing-lesson.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class FishingLessonService {

  baseURL = "http://localhost:8080/";

  constructor(private http: HttpClient) { }

  getAuthoHeader() : any {
    const headers = {
      'Content-Type': 'application/json',
      'Authorization' : 'Bearer ' + localStorage.getItem("token")
    }
    return{
      headers: headers
    };
  } 

  createFishingLesson(data: any){
    return this.http.post(this.baseURL + "api/fishing/createFishingLesson", data, this.getAuthoHeader());
  }

  getByInstructorId(id: any){
    return this.http.get(this.baseURL + "api/fishing/fishingLessons/" + id, this.getAuthoHeader());
  }

  getById(id: any){
    return this.http.get(this.baseURL + "api/fishing/fishingLesson/" + id, this.getAuthoHeader());
  }

  getByIdForInstructor(id: any){
    return this.http.get(this.baseURL + "api/fishing/instructorFishingLesson/" + id, this.getAuthoHeader());
  }

  editFishingLesson(data: any){
    return this.http.put(this.baseURL + "api/fishing/editFishingLesson", data, this.getAuthoHeader());
  }

  getAll(){
    return this.http.get(this.baseURL + "api/fishing/fishingLessons");
  }

  getOneLesson(id: any){
    return this.http.get(this.baseURL + "api/fishing/lessonById/" + id);
  }

  sort(data: any){
    return this.http.post(this.baseURL + "api/fishing/sort" ,  data);
  }

  sortAvailable(data: any){
    return this.http.post(this.baseURL + "api/fishing/sortAvailable" ,  data, this.getAuthoHeader());
  }

  search(searchTerm: any){
    return this.http.get(this.baseURL + "api/fishing/search/" + searchTerm);
  }

  searchForInstructor(searchTerm: any, id: any){
    return this.http.get(this.baseURL + "api/fishing/searchForInstructor/" + searchTerm + "/" + id, this.getAuthoHeader());
  }

  getAvailableLessons(data: any){
    return this.http.post(this.baseURL + "api/fishing/availableLessons" ,  data, this.getAuthoHeader());
  }

  getAvailableLessonsForInstructor(data: any){
    return this.http.post(this.baseURL + "api/fishing/availableLessonsForInstructor" ,  data, this.getAuthoHeader());
  }

  getAvailableBoatsForCertainDate(data: any){
    return this.http.post(this.baseURL + "api/fishing/availableLessonsForCertainDate" ,  data, this.getAuthoHeader());
  }

  subscribedLessons(id: any){
    return this.http.get(this.baseURL + "api/fishing/subscribedLessons/" + id, this.getAuthoHeader());
  }

  searchByManyParams(data: any){
    return this.http.post(this.baseURL + "api/fishing/searchByManyParams" ,  data, this.getAuthoHeader());
  }

  deleteLesson(id: any){
    return this.http.get(this.baseURL + "api/fishing/delete/" + id, this.getAuthoHeader());
  }

}
